lightning/src/ln/channelmanager.rs

```diff
@@ -437,13 +439,45 @@ pub struct ChannelManager<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref,
 	/// Used when we have to take a BIG lock to make sure everything is self-consistent.
 	/// Essentially just when we're serializing ourselves out.
 	/// Taken first everywhere where we are making changes before any other locks.
-	total_consistency_lock: RwLock<()>,
+	/// When acquiring this lock in read mode, rather than acquiring it directly, call
+	/// `PersistenceNotifierGuard::new(..)` and pass the lock to it, to ensure the PersistenceNotifier
+	/// the lock contains sends out a notification when the lock is released.
+	total_consistency_lock: RwLock<PersistenceNotifier>,
```

Comment:
Hmm, combining these is somewhat annoying because it required re-introducing the Arc (and, worse, regular clones of the Arc instead of it never being touched). It is a nicer design, but maybe drop the Arc and just hold the notifier lock through the notify call in `PersistenceNotifierGuard::drop` - its relatively unlikely that you'd hit the
```
notified ->
lock ->
block
                      unlock
run
```
pessimal case because thread 1 immediately goes to drop its lock (after a return) after notify and thread 2 has to check the persistence_lock boolean before trying (after a call) to take the lock. Its certainly not guaranteed, but to avoid the extra arc clone every time we drop the notifier its probably worth it.